.github/workflows/main.yml

```diff
@@ -95,19 +97,51 @@ jobs:
           rm codeql-home/codeql.zip
         shell: bash
 
-      - name: Run unit tests (Linux)
+      - name: Get Nightly Release URL
+        id: get-url
+        env:
+          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'
+        shell: bash
+        # This workflow step gets an unstable testing version of the CodeQL CLI. It should not be used outside of these tests.
+        run: |
+          export LATEST=`gh release list --repo dsp-testing/codeql-cli-nightlies -L 1 | cut -f 3`
+          echo "::set-output name=nightly-url::https://github.com/dsp-testing/codeql-cli-nightlies/releases/download/$LATEST/codeql-bundle.tar.gz"
+
+      - name: Install Nightly CodeQL
+        run: |
+          mkdir codeql-nightly-home
+          cd codeql-nightly-home
+          curl -L --silent ${{ steps.get-url.outputs.nightly-url }} -o codeql.tar.gz
+          tar -xf codeql.tar.gz
+          rm codeql.tar.gz
+        shell: bash
+
+      - name: Run unit tests with stable CodeQL (Linux)
         working-directory: extensions/ql-vscode
         if: matrix.os == 'ubuntu-latest'
         run: |
           CODEQL_PATH=$GITHUB_WORKSPACE/codeql-home/codeql/codeql npm run test
 
-      - name: Run unit tests (Windows)
+      - name: Run unit tests with nightly CodeQL (Linux)
```

Comment:
We could add the CLI version (stable / nightly) to the job matrix, but given the short duration of these steps, that probably wouldn't be worth the extra compute.

Reply:
Hmm, it's probably still worth doing to make the workflow a bit neater though.